### ISSUE TITLE
CMake on OS X not producing Xcodeproj file possibly due to <malloc.h> not being available?

### ISSUE THREAD

**Issue by msmollin** [CorsixTH/CorsixTH#961]
I'm attempting to build the OS X version of CorsixTH from source (see #959 & #960 for other issues) and assuming I satisfy 959 and 960, I can successfully configure the project, and click the generate button.

However generation goes very quickly (even for the computer I'm working on) and no xcodeproj bundle is produced. Looking in CMakeError.log reveals what I believe is the culprit:

```
Determining if files malloc.h exist failed with the following output:
Change Dir: /Users/msmollin/corsix_build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_bd4de/fast"
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/cmTC_bd4de.dir/build.make CMakeFiles/cmTC_bd4de.dir/build
Building C object CMakeFiles/cmTC_bd4de.dir/CheckIncludeFiles.c.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc    -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -mmacosx-version-min=10.11   -o CMakeFiles/cmTC_bd4de.dir/CheckIncludeFiles.c.o   -c /Users/msmollin/corsix_build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c
/Users/msmollin/corsix_build/CMakeFiles/CMakeTmp/CheckIncludeFiles.c:2:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^
1 error generated.
make[1]: *** [CMakeFiles/cmTC_bd4de.dir/CheckIncludeFiles.c.o] Error 1
make: *** [cmTC_bd4de/fast] Error 2

Source:
/* */
#include <malloc.h>


int main(void){return 0;}
```

This is an OS X specific problem: since OS X based on FreeBSD 7 or 8, malloc.h is actually defined in stdlib.h. Other projects have run into this problem and ifdef'd around it (see - https://github.com/facebook/folly/issues/116), but I am loathe to make this change by myself due to my unfamiliarity with this codebase. Any thoughts on how to proceed? Is this actually the error that I'm running into, or is it possibly because I solved #959 & #960 incorrectly?

I wish github had a way to link issues as dependencies!

**Comment by TheCycoONE** [comment]
I believe malloc.h should be removed entirely (replaced by cstdlib), it's not a standard header. You can try it in a PR and if it still builds with travis / appveyor then it can be merged.

**Comment by TheCycoONE** [comment]
It's a bit harder to remove then that. We use alloca from malloc.h. That usage should be and can be replaced with standard C++ 11 code so I will work on it.

**Comment by msmollin** [comment]
Thanks for looking at this, @TheCycoONE. I figured it was probably more involved than just changing it to stdlib.h (since using cstdlib I think changes how you call those functions into a namespace with a prefix of `std::`). I'll work on getting the other PR's in for the other two issues and then maybe we can see if we can resolve this guy.

**Comment by msmollin** [comment]
@tobylane have you noticed the xcodeproj bundle not being produced? I know the travis project manually calls xcodebuild but I figured it would hit this as well.

**Comment by tobylane** [comment]
It works for me locally just as it does on Travis. In terms of malloc being required, it appears to not find it and use alloca instead (https://travis-ci.org/tobylane/CorsixTH/jobs/99822377 line 2834) and works after 963 (https://travis-ci.org/tobylane/CorsixTH/jobs/100076227 same lines don't appear at 2837). Both builds can be opened.

**Comment by TheCycoONE** [comment]
Should be resolved by #963
